### ISSUE TITLE
Support separate, accessible labels for Nimble components

### ISSUE THREAD

**Issue by m-akinc** [ni/nimble#727]
## 😯 Problem to Solve

There are use cases where a client may wish to not use a control's built-in label (if exists), but rather provide a separate label element for the control. This is currently the only way to label a select or combobox. In such use cases, we still want the control to have an accessible name. There should be some clear pattern (that we can document) for achieving this, e.g. through the use of `aria-label`, `aria-labelledby`, or `<label for="...">`.

## 💁 Proposed Solution

We should look at each of our controls and ensure there is some way to give the control an accessible name, even when not using the standard, built-in label. We should try to use the same pattern for all controls. Whatever approach we come up with should be documented in `nimble-components\docs\accessibility.md`.

**Comment by rajsite** [comment]
> An additional consideration is to automatically apply an aria label to a button if it has `content-hidden` and provides a title.

This should already be the case for buttons, ie for the example: https://nimble.ni.dev/storybook/?path=/story/button--outline-button&args=label:hello+world;icon:true;contentHidden:true

which has code snippet:
```html

<nimble-button appearance="outline" content-hidden>
    <nimble-icon-key slot="start"></nimble-icon-key>
    hello world
</nimble-button>
```

The button should still be labelled by the button text in the accessibility tree even though it renders as an icon:

![image](https://user-images.githubusercontent.com/1588923/201740042-fd0b3dda-f4f0-4a00-a70d-9f313fdecaac.png)

**Comment by TrevorKarjanis** [comment]
> This should already be the case for buttons, ie for the example: https://nimble.ni.dev/storybook/?path=/story/button--outline-button&args=label:hello+world;icon:true;contentHidden:true

Yes, Molly pointed that out, and I am using it. I'm just wondering if it would be worth the convenience when the button has a title to not have to specify it twice, once for the title and once as hidden text.

**Comment by rajsite** [comment]
> Yes, Molly pointed that out, and I am using it. I'm just wondering if it would be worth the convenience when the button has a title to not have to specify it twice, once for the title and once as hidden text.

Conceptually we are treating those separately, the title would be more informational / descriptive and the label is the button label. ie "Submit" as the button label (what the icon represents if only an icon is visible) vs the title may be "create a new thingie".

[Title itself is unfortunately not accessible](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns) so the long-term approach would be assigning a tooltip to the button for that descriptive text, but we don't have that yet.

But at least currently the mindset is that the title/tooltip is different from the label name.
